arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/createTests/advanced/Scenario4UnitTests.java

```diff
@@ -0,0 +1,143 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.zktracer.instructionprocessing.createTests.advanced;
+
+import static net.consensys.linea.zktracer.instructionprocessing.createTests.advanced.ScenarioUtils.*;
+import static net.consensys.linea.zktracer.instructionprocessing.utilities.MonoOpCodeSmcs.userAccount;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import net.consensys.linea.reporting.TracerTestBase;
+import net.consensys.linea.testing.ToyExecutionEnvironmentV2;
+import net.consensys.linea.testing.TransactionProcessingResultValidator;
+import net.consensys.linea.zktracer.instructionprocessing.utilities.SmartContractTestValidator;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+// Recap : CustomCreate2 is a contract with multiple methods to deploy ContractC using CREATE2
+// opcode
+// See more details in AllScenariiInitCodeTests
+
+/*
+
+SCENARIO 1 - ATTEMPT CREATE2 AFTER A CREATE2
+
+Attempts a create2 after a successful create2.
+Done by calling the ContractC that has just been deployed to do a callback to CustomCreate2 and attempt redeployment.
+
+       CALL
+     -------->  (1) CREATE2 -----> ContractC deployed
+                     CALLC                CALL
+                (2) ---------> callback ---------> CREATE2 -----> Collision, deployment fails
+
+When Nested :
+
+       CALL           CALL               CALL               CALL
+      -------->  (0) -------> Scenario3 -------> Scenario2 -------> Scenario1
+                 (1) and (2) stay the same
+
+Note : storeInitCodeC and storeSalt transactions are preparation transactions
+
+*/
+
+public class Scenario4UnitTests extends TracerTestBase {
+
+  /*
+  SCENARIO 4 -
+  TXSTATUS : Successful
+  LOGS: 1 ContractCreated, 1 CallCreate2WithInitCodeC_noValue
+  Note: transaction is sent with value 2
+  Note 2 : used in ScenariiTriggeredFromRoot
+   */
+  @Test
+  void deployScenario4(TestInfo testInfo) {
+    Map<String, List<Integer>> logsTopicMap = new HashMap<>();
+    List<Integer> txStatuses = List.of(1, 1, 1);
+
+    logsTopicMap.put(contractCreatedEvent, List.of(0, 0, 1));
+    // Call to attempt a redeployment after the successful one
+    logsTopicMap.put(callCreate2WithInitCodeC_noValue_Event, List.of(0, 0, 1));
+
+    // Instantiate validator
+    TransactionProcessingResultValidator create2OneTxValidator =
+        new SmartContractTestValidator(txStatuses, logsTopicMap, new HashMap<>());
+
+    List<Transaction> transactions =
+        getTransactions(
+            customCreate2Account,
+            userAccount,
+            List.of(storeInitCodeC, storeSalt, create2WithCallCtoCallback_noValue),
+            List.of(0L, 0L, 2L));
```

Comment:
As a reader (and future reader) it would be nice to have, as the final argument, a list of things rather than integers. E.g. things that implement an interface with a `public int value()` method or so used to insert the actual values into the transaction builder. So that the final list could be e.g.
```java
List.of(NONE, NONE, CREATE2_WITH_CALL_BACK)
```
where e.g. you have
```java
enum AdvancedCreate2TestConstructorScenario implements ScenarioValue {
    ...
    CREATE2_WITH_CALL_BACK(2),
    SELFDESTRUCT(3),
    REVERT(4);

private final int value;

// constructor

// ScenarioValue method implementation returning value

}
```